plugin/trino-hive/src/main/java/io/trino/plugin/hive/avro/AvroHiveFileWriterFactory.java

```diff
@@ -0,0 +1,143 @@
+import io.trino.spi.connector.ConnectorSession;
+import io.trino.spi.type.Type;
+import io.trino.spi.type.TypeManager;
+import org.apache.avro.Schema;
+
+import java.io.Closeable;
+import java.util.List;
+import java.util.Optional;
+import java.util.OptionalInt;
+import java.util.Properties;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static io.trino.memory.context.AggregatedMemoryContext.newSimpleAggregatedMemoryContext;
+import static io.trino.plugin.hive.HiveErrorCode.HIVE_WRITER_OPEN_ERROR;
+import static io.trino.plugin.hive.HiveMetadata.PRESTO_QUERY_ID_NAME;
+import static io.trino.plugin.hive.HiveMetadata.PRESTO_VERSION_NAME;
+import static io.trino.plugin.hive.HiveSessionProperties.getTimestampPrecision;
+import static io.trino.plugin.hive.HiveSessionProperties.isAvroNativeWriterEnabled;
+import static io.trino.plugin.hive.util.HiveClassNames.AVRO_CONTAINER_OUTPUT_FORMAT_CLASS;
+import static io.trino.plugin.hive.util.HiveUtil.getColumnNames;
+import static io.trino.plugin.hive.util.HiveUtil.getColumnTypes;
+import static java.util.Objects.requireNonNull;
+
+public class AvroHiveFileWriterFactory
+        implements HiveFileWriterFactory
+{
+    private final TrinoFileSystemFactory fileSystemFactory;
+    private final TypeManager typeManager;
+    private final NodeVersion nodeVersion;
+
+    @Inject
+    public AvroHiveFileWriterFactory(
+            TrinoFileSystemFactory fileSystemFactory,
+            TypeManager typeManager,
+            NodeVersion nodeVersion)
+    {
+        this.fileSystemFactory = requireNonNull(fileSystemFactory, "hdfsEnvironment is null");
+        this.typeManager = requireNonNull(typeManager, "typeManager is null");
+        this.nodeVersion = requireNonNull(nodeVersion, "nodeVersion");
+    }
+
+    @Override
+    public Optional<FileWriter> createFileWriter(
+            Location location,
+            List<String> inputColumnNames,
+            StorageFormat storageFormat,
+            HiveCompressionCodec compressionCodec,
+            Properties schema,
+            ConnectorSession session,
+            OptionalInt bucketNumber,
+            AcidTransaction transaction,
+            boolean useAcidSchema,
+            WriterKind writerKind)
+    {
+        if (!isAvroNativeWriterEnabled(session)) {
+            return Optional.empty();
+        }
+        if (!AVRO_CONTAINER_OUTPUT_FORMAT_CLASS.equals(storageFormat.getOutputFormat())) {
+            return Optional.empty();
+        }
+        checkArgument(!useAcidSchema, "Acid is not supported");
+
+        AvroCompressionKind compressionKind = compressionCodec.getAvroCompressionKind().orElse(AvroCompressionKind.NULL);
+        if (!compressionKind.isSupportedLocally()) {
+            throw new TrinoException(HIVE_WRITER_OPEN_ERROR, "Avro Compression codec %s is not supported in the environment".formatted(compressionKind));
```

Comment:
This should probably be `VerifyException` since Trino should always support all codecs. It isn't something specific to the environment or user. It would be a bug or packaging issue on our side. So you could use `verify()` here.

Reply:
Will do. I think it a packaging issue on Avro side. Which changes if we insert our own CodecFactories

Reply:
Like we may not longer need this if we put in out own.